make/targets/golang/version.mk

```diff
@@ -0,0 +1,61 @@
+self_dir :=$(dir $(lastword $(MAKEFILE_LIST)))
+
+.empty-golang-versions-files:
+	@mkdir -p "$(PERMANENT_TMP)"
+	@rm -f "$(PERMANENT_TMP)/golang-versions" "$(PERMANENT_TMP)/named-golang-versions"
+	@touch "$(PERMANENT_TMP)/golang-versions" "$(PERMANENT_TMP)/named-golang-versions"
+.PHONE: .empty-golang-versions-files
+
+verify-golang-versions:
+	@if [ -f "$(PERMANENT_TMP)/golang-versions" ]; then \
+		LINES=$$(cat "$(PERMANENT_TMP)/golang-versions" | sort | uniq | wc -l); \
+  		if [ $${LINES} -gt 1 ]; then \
+			echo "Golang version mismatch:"; \
+			echo; \
+			cat "$(PERMANENT_TMP)/named-golang-versions"; \
+			false; \
+    	fi; \
+    fi
+.PHONY: verify-golang-versions
+
+# $1 - filename (symbolic, used as postfix in Makefile target)
+# $2 - golang version
+define add-golang-version-reference-internal
+verify-golang-versions-$(1): .empty-golang-versions-files
+verify-golang-versions-$(1):
+	@echo "$(1): $(2)" >> "$(PERMANENT_TMP)/named-golang-versions"
+	@echo "$(2)" >> "$(PERMANENT_TMP)/golang-versions"
+.PHONY: verify-golang-versions-$(1)
+
+verify-golang-versions: verify-golang-versions-$(1)
+.PHONY: verify-golang-versions
+endef
+
+# $1 - filename (symbolic, used as postfix in Makefile target)
+# $2 - golang version
+define add-golang-version-reference
+$(eval $(call add-golang-version-reference-internal,$(1),$(2)))
+endef
+
+# $1 - Dockerfile filename (symbolic, used as postfix in Makefile target)
+define add-Dockerfile-builder-golang-version
+$(call add-golang-version-reference,$(1),$(shell grep "AS builder" "$(1)" | sed 's/.*golang-\([[:digit:]][[:digit:]]*.[[:digit:]][[:digit:]]*\).*/\1/'))
+endef
+
+define add-go-mod-golang-version
+$(call add-golang-version-reference,go.mod,$(shell grep -e 'go [[:digit:]]*\.[[:digit:]]*' go.mod | sed 's/go //'))
+endef
+
+define add-buildroot-golang-version
+$(call add-golang-version-reference,.ci-operator.yaml,$(shell grep -e 'tag: golang-[[:digit:]]*\.[[:digit:]]*' .ci-operator.yaml | sed 's/.*tag: golang-//'))
```

Comment:
(The `grep | sed` can be combined by using `sed -n 's/…/…/p`, maybe using something like `sed -n 's/.*tag: golang-\([[:digit:]]*\.[[:digit:]]*.*\)/\1/p'`. That way the regex can’t get out of sync between the two statements, OTOH it’s using less known options.)